src/compiler/checker/lines_8000_12000.rs

```diff
@@ -88,17 +88,27 @@ impl TypeChecker {
         if let Some(declared_type) = declared_type {
             return Some(self.add_optionality(declared_type, Some(is_property), Some(is_optional)));
         }
-        unimplemented!()
+
+        if has_only_expression_initializer(declaration)
+            && declaration
+                .as_has_expression_initializer()
+                .maybe_initializer()
+                .is_some()
+        {
+            let type_ = self.check_declaration_initializer(declaration, None);
```

Comment:
Here's where we're inferring the type of a variable declaration based on the initializer when it didn't have an explicit type annotation